### PR TITLE
Split content to files and exclude

### PR DESCRIPTION
**Pull request by yufeih**
#3513 

Split `content` to `files` and `exclude`, after this change the only supported `oneof` syntax is _token_ or _token array_

cc @LucasDvp @mingwli

### CODE REVIEW

File: src/docfx/config/Config.cs [hunk 1]
```diff
@@ -23,9 +26,14 @@ internal sealed class Config
...
-        /// Gets the contents that are managed by this docset.
+        /// Gets the file glob patterns included by the docset.
+        /// </summary>
+        public readonly string[] Content = s_defaultContentInclude;
+
+        /// <summary>
+        /// Gets the file glob patterns excluded from this docset.
         /// </summary>
-        public readonly FileConfig Content = new FileConfig();
+        public readonly string[] ContentExclude = s_defaultContentExclude;
```
Comment (superyyrrzz) [inline src/docfx/config/Config.cs]: Naming: what about `files` and `exclude` as in v2? Jekyll also has a `exclude` at config's top level.

Comment (yufeih) [reply src/docfx/config/Config.cs]: sure

File: src/docfx/config/Config.cs [hunk 2]
```diff
@@ -23,9 +25,14 @@ internal sealed class Config
         public readonly string Name = string.Empty;
 
         /// <summary>
-        /// Gets the contents that are managed by this docset.
+        /// Gets the file glob patterns included by the docset.
+        /// </summary>
+        public readonly string[] Files = s_defaultInclude;
```
Comment (OsmondJiang) [inline src/docfx/config/Config.cs]: how about
```
content:
  include:
  exclude:
```

Comment (OsmondJiang) [reply src/docfx/config/Config.cs]: or use `!` to represent exclude, only `files`
```
files:
  - a.md
  - !b.md
```

**Comment by yufeih** [comment]
Test failed due to configs in dependent repos, @superyyrrzz @OsmondJiang , I'll continue this PR when you guys finished migrating test cases.

File: src/docfx/config/Config.cs (hunk 2 above)
Comment (yufeih) [reply src/docfx/config/Config.cs]: The final schema is `files` and `exclude`.

**Pull request merged**